### DISCUSSION TITLE
unit testing with php 7.4 and php 8

### DISCUSSION THREAD

**Discussion by Ramoonus** [pods-framework/pods#6786] [Ideas]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

[a comment by an automated account is not shown]

**Comment by Ramoonus** [comment]
Need a patch?

**Comment by Ramoonus** [comment]
hot-testing https://travis-ci.org/github/pods-framework/pods/builds/666267925

**Comment by JoryHogeveen** [comment]
Hi @Ramoonus 
Do you have a patch ready for PHP 7.4 support?

**Comment by JoryHogeveen** [comment]
@sc0ttkclark I've added this as a blocker issue to verify before releasing the next maintenance release. I want to make sure it fully supports and is tested with PHP 7.4 before releasing since our release interval is quite long.

**Comment by Ramoonus** [comment]
Not yet
Phpunit 4 is the current bottleneck

**Comment by sc0ttkclark** [comment]
FYI, Pods 2.8 runs on Codeception right now which uses PHPUnit but probably a version that's not PHPUnit 9.

**Comment by Ramoonus** [comment]
Will check

**Comment by Ramoonus** [comment]
https://github.com/pods-framework/pods/blob/2.x/composer.json#L61

**Comment by Ramoonus** [comment]
https://github.com/Codeception/Codeception/releases/tag/4.1.0
should work

**Comment by JoryHogeveen** [comment]
@sc0ttkclark Punt this to be done for 2.8?

**Comment by Ramoonus** [comment]
need help with 8.x too?

**Comment by JoryHogeveen** [comment]
Sure, thanks!

**Comment by Ramoonus** [comment]
i need to check how to separate phpunit per build

**Comment by sc0ttkclark** [comment]
Pods 2.8 testing is using a strict set of versions right now that are verified to work:

* wp-browser 2.2.15 (via composer)
* Codeception 2.5.6 (via composer)
* PHPUnit 6.5.14 (via composer)
* PHP 7.3.8 (via [tric](https://github.com/the-events-calendar/tric))

Updating PHP will be a different kind of process that we'll have to be careful to test as we push Codeception / wp-browser to a newer version.
